custom_components/tuya_local/devices/smart_wifi_socket_Thermostat.yaml

```diff
@@ -0,0 +1,279 @@
+products:
+  - id: msclnsfe0q3yd39q
+primary_entity:
+  entity: climate
+  translation_key: thermostat
+  dps:
+    - id: 4
+      type: string
+      name: hvac_mode
+      mapping:
+        - dps_val: "heat"
+          value: "heat"
+        - dps_val: "cool"
+          value: "cool"
+        - dps_val: "plug"
+          value: "off"
+    - id: 6
+      type: integer
+      name: current_temperature
+      mapping:
+        - scale: 10
+    - id: 9
+      type: integer
+      name: max_temperature
+      optional: true
+      mapping:
+        - scale: 10
+    - id: 12
+      type: integer
+      name: min_temperature
+      optional: true
+      mapping:
+        - scale: 10
+    - id: 17
+      type: integer
+      name: temperature
+      unit: C
+      mapping:
+        - scale: 10
+    - id: 114
+      type: integer
+      name: hvac_action
+      mapping:
+        - dps_val: 0
+          constraint: hvac_mode
+          conditions:
+            - dps_val: false
+              value: "off"
+            - dps_val: true
+              value: idle
+        - dps_val: 1
+          constraint: hvac_mode
+          conditions:
+            - dps_val: heat
+              value: heating
+            - dps_val: cool
+              value: cooling
+secondary_entities:
+  # Control
+  # Control relay in plug mode
+  - entity: switch
+    name: Relay Control
+    dps:
+      - id: 2
+        name: switch
+        type: boolean
+  # Turn on the display on press for timeout time
+  - entity: button
+    name: Display On
+    optional: true
+    dps:
+      - id: 118
+        name: button
+        type: boolean
+        
+
+  # Diagnostics
+  - entity: binary_sensor
+    class: power
+    category: diagnostic
+    name: Relay status
+    dps:
+      - id: 114
+        type: integer
+        name: sensor
+    mapping:
+      - dps_val: 0
+        value: false
+      - dps_val: 1
+        value: true
+
+  # Configuration
+  # Child Lock
+  - entity: lock
+    translation_key: child_lock
+    category: config
+    dps:
+      - id: 110
+        name: lock
+        type: boolean
+
+  # Display Timeout
+  - entity: select
+    name: Display Timeout
+    icon: "mdi:timer-cog-outline"
+    category: config
+    optional: true
+    dps:
+      - id: 117
+        type: string
+        name: option
+        mapping:
+          - dps_val: "15s"
+            value: "15s"
+          - dps_val: "30s"
+            value: "30s"
+          - dps_val: "60s"
+            value: "60s"
+          - dps_val: "120s"
+            value: "120s"
+
+  # High temperature alarm audio
+  - entity: switch
+    category: config
+    name: Audio Alarm High temperature
+    icon: "mdi:thermometer-alert"
+    dps:
+      - id: 8
+        name: switch
+        type: boolean
+  # High temperature alarm setting
+  - entity: number
+    name: Alarm Maximum temperature
+    category: config
+    icon: "mdi:thermometer-chevron-up"
+    optional: true
+    dps:
+      - id: 9
```

Comment:
Correct, but the audio alarm can be turned on/off. Setting the setpoint outside or on the alarm value is a bit odd so thought this was a nice way to give the user freedom to set the high and low for the thermostat (being in cooling or heating mode). Setting the alarm values would limit the range.

Reply:
And thanks for all the feedback :)

Reply:
The main problem with this is that an alarm threshold will let you set a value that is outside the range that the actual thermostat can be set to. If you use it as max_temperature, then it will override the range of the thermometer. It makes sense when it is within the range, but not outside it (unfortunately we do not have a way to specify that the minimum of two values should be used).

Reply:
Thermometer goes -30 to 120 and that's what the alarm can be set to. The thermostat range is -30 to 110. There are dps linked to it but, they are read only. Using those as the min/max makes the graphic in HA rather useless as it is impossible to set it correctly from there. 
Using the alarm values gives boundaries to the thermostat as well. Using the above 110 values sounds weird while the negative could be used for a freezer or something (I'm already using one to fix a broken refrigerator for example).

If it would be possible to make dummy DPs I could use 2 of those to set the min/max

Reply:
I think it is OK as it is based on your first sentence. But later you mention 110 as the upper limit. If the alarm limit and thermostat limit are different, it may be better to limit to the lower of those to avoid unexpected behaviour due to setting the temperature out of range.  Personally I find it hard to think of any practical uses above 100, as for normal domestic use, mostly either water or air temperatures are going to be monitored.